runtime/lua/vim/diagnostic.lua

```diff
@@ -865,7 +865,16 @@ local function set_list(loclist, opts)
     end
   end
   if open then
-    api.nvim_command(loclist and 'lwindow' or 'botright cwindow')
+    if loclist then
+      api.nvim_command('lwindow')
+    else
+      -- First navigate to the diagnostics quickfix list.
+      local nr = vim.fn.getqflist({ id = _qf_id, nr = 0 }).nr
```

Comment:
It should be `qf_id` after #31486

Reply:
Oops, see https://github.com/neovim/neovim/pull/31557